lib/iris/cube.py

```diff
@@ -879,22 +889,27 @@ def _add_unique_aux_coord(self, coord, data_dims):
             data_dims = (int(data_dims),)
 
         if data_dims:
-            if len(data_dims) != coord.ndim:
+            if len(data_dims) != metadata.ndim:
                 msg = 'Invalid data dimensions: {} given, {} expected for ' \
-                      '{!r}.'.format(len(data_dims), coord.ndim, coord.name())
+                      '{!r}.'.format(len(data_dims), metadata.ndim,
+                                     metadata.name())
                 raise ValueError(msg)
             # Check compatibility with the shape of the data
             for i, dim in enumerate(data_dims):
-                if coord.shape[i] != self.shape[dim]:
+                if metadata.shape[i] != self.shape[dim]:
                     msg = 'Unequal lengths. Cube dimension {} => {};' \
-                          ' coord {!r} dimension {} => {}.'
+                          ' metadata {!r} dimension {} => {}.'
                     raise ValueError(msg.format(dim, self.shape[dim],
-                                                coord.name(), i,
-                                                coord.shape[i]))
-        elif coord.shape != (1,):
+                                                metadata.name(), i,
+                                                metadata.shape[i]))
+        elif metadata.shape != (1,):
             raise ValueError('Missing data dimensions for multi-valued'
-                             ' coordinate {!r}'.format(coord.name()))
+                             ' {} {!r}'.format(mtype, metadata.name()))
```

Comment:
@marqh I'd prefer that we didn't pass the `mtype` string into this method just to be used as part of a message in the exception.

You could replace the use of `mtype` with, for example `type(metadata).__name__` or `metadata.__class__.__name__`